test/test_fcntl.rs

```diff
@@ -52,12 +52,42 @@ mod linux_android {
 
     use libc::loff_t;
 
-    use nix::fcntl::{SpliceFFlags, FallocateFlags, fallocate, splice, tee, vmsplice};
+    use nix::fcntl::{SpliceFFlags, FallocateFlags, CopyFileRangeFlags, fallocate, splice, tee, vmsplice, copy_file_range};
     use nix::sys::uio::IoVec;
     use nix::unistd::{close, pipe, read, write};
 
     use tempfile::{tempfile, NamedTempFile};
 
+    /// This test creates a temporary file containing the contents
+    /// 'foobarbaz' and uses the `copy_file_range` call to transfer
+    /// 3 bytes at offset 3 (`bar`) to another empty file at offset 0. The
+    /// resulting file is read and should contain the contents `bar`.
+    /// The from_offset should be updated by the call to reflect
+    /// the 3 bytes read (6).
+    #[test]
+    fn test_copy_file_range() {
+        const CONTENTS: &[u8] = b"foobarbaz";
+
+        let mut tmp1 = tempfile().unwrap();
+        let mut tmp2 = tempfile().unwrap();
+
+        tmp1.write_all(CONTENTS).unwrap();
+        tmp1.flush().unwrap();
+
+        let mut from_offset: libc::loff_t = 3;
+        copy_file_range(tmp1.as_raw_fd(), Some(&mut from_offset),
+                                    tmp2.as_raw_fd(), None,
+                                    3, CopyFileRangeFlags::NONE).unwrap();
+
+        tmp2.sync_all().unwrap();
```

Comment:
I the `sync_all` necessary?  I would think that the test should work fine without it.

Reply:
Apparently not. I was trying to be overly cautious.